### ISSUE TITLE
opam2 2.0.0~beta2 destroyed my MANPATH env var (I use bash)

### ISSUE THREAD

**Issue by UnixJunkie** [ocaml/opam#2870]
# echo $MANPATH
    :/home/me/.opam/ocaml-base-compiler.4.04.0/man

then 'man grep' doesn't work ...

I think it would have been better to leave my MANPATH untouched.

**Comment by AltGr** [comment]
What system ? This is surprising, I indeed have `MANPATH` set to a similar value here, but `man` works fine with it.
Was the variable set beforehand ?

Note that you can now try `eval $(opam env --revert)`

**Comment by UnixJunkie** [comment]
CentOS release 6.8.
The variable was unset beforehand.

**Comment by mato** [comment]
@AltGr I've just discovered the same problem with `2.0.0~rc`. 

I'm on Debian 9, using the default empty setting of `MANPATH`. It appears that if `opam env` includes a `MANPATH` then _only_ those directories get searched for man pages.

This matches the _ENVIRONMENT_ section in the docs for `man`:

> ENVIRONMENT
>       MANPATH
>              If  $MANPATH is set, its value is used as the path to search for manual pages.

Given that there does not appear to be a way to determine the default value for `MANPATH`, I think this "feature" should be removed. Presumably it's new in OPAM 2, as I've not noticed this with 1.x.

**Comment by mato** [comment]
Trying to work around this with `eval (opam env | grep -v ^MANPATH)` kind of works, but opam seems to check that _all_ the things in `opam env` were set, so it still complains that the environment is out of sync.

**Comment by mato** [comment]
Ok, this is weird -- I just set everything back the way it was, restarted some shells and now it all works again with the full `eval $(opam env)`. Unfortunately I don't have the original shell that caused the problem open anymore, so I don't know what the opam-set MANPATH that triggered it is.

However, experimenting manually, it appears that at least on my Debian system, if I set MANPATH to `:/some/path` then that _does_ get appended to the system MANPATH. If I leave out the leading `:`, then I get the "only search what's in MANPATH" behaviour.

I'll report back if I manage to reproduce this again; there's definitely some kind of bug here.

**Comment by AltGr** [comment]
Ok, thanks

**Comment by AltGr** [comment]
>  Presumably it's new in OPAM 2, as I've not noticed this with 1.x.

I'll need to double-check, but I am pretty sure it was there since the beginning... We disabled it for openBSD IIRC due to a similar issue but the leading `:` hadn't caused issues on Linux so far.

**Comment by UnixJunkie** [comment]
The leading ':' looks necessary on Linux to mean "append the following to the system-wide manpath"

**Comment by UnixJunkie** [comment]
It would be nice if some current centos user could report if it works nowadays with a recent opam.

**Comment by blerner** [comment]
I believe I'm seeing something similar, on Ubuntu 18.04 + opam 2.0.5 + fish shell.  As with the original post, `man grep` doesn't work, and neither does anything else...  After some digging, I've found:

- My `.config/fish/config.fish` file contained `source /home/blerner/.opam/opam-init/init.fish > /dev/null 2> /dev/null or true`
- `.../.opam/opam-init/init.fish` sources `env_hook.fish` and `variables.fish`
- Commenting out that source line avoids setting `MANPATH` at all, which gets `man` working again.  But of course my OPAM variables are not present.  So I don't want to get stuck with this partial solution"
- I tried commenting out the line in `variables.fish` that sets `MANPATH`, but somehow `MANPATH` still gets set, /nothing/ seemed to unset it, including `set -e MANPATH` and `env -u MANPATH`, and I couldn't even set it to another value.
- I just found this bug report, and saw the comment about `env_hook.fish`.  From there, I grepped for MANPATH within the .opam directory, and found it in `.opam/default/.opam-switch/environment`
- Once I commented out the MANPATH line in `environment`, things started working again.
- Nothing else in my configuration files ever sets the `MANPATH` variable; any changes are solely coming from OPAM.
- If I manually set an environment variable such that it has a leading colon, then `man` works properly.  So it seems that `env_hooks` isn't respecting leading colons correctly, or possibly it's not attaching a leading colon *when the environment variable is otherwise empty*.

It appears the relevant code is [here](https://github.com/ocaml/opam/blob/2c513841184ee4ed718bb2bef4fd22e7b5c2af86/src/state/opamEnv.ml#L46-L63), but I don't quite understand the usage of `apply_op_zip` well enough to see whether it's preserving colons properly...
